pkgs/development/compilers/llvm/common/tblgen.nix

```diff
@@ -60,6 +61,17 @@ let
     src = src';
     sourceRoot = "${src.name}/llvm";
 
+    postPatch = ''
+      (
+        cd ../clang
+        chmod u+rwX -R .
+        for p in ${toString clangPatches}
```

Comment:
I'm allergic to `toString` as a mechanism of forming bash arrays. I see it's quite common across nixpkgs, though.

Reply:
Thanks for the review -- any suggestion of how to express this differently? I'm also highly aware of shell expansion, but my belief (correct me if I'm wrong please) is that these paths should not have spaces in them.  If they can have spaces then it's worth knowing, even if there isn't an obvious better thing to write in this context.

Reply:
I think it's definitely safe with respect to spaces, assuming everything in the list is a Nix store path. What I dislike about `toString` is how it behaves with other Nix datatypes, and how it doesn't express the constraints of taking a list of things and rendering it in bash so that it could be iterated over.

My ideal replacement -- which doesn't exist, as far as I'm aware -- would assert things about the incoming Nix values, and only then land them into the string.